### PR TITLE
[Refactor]: View.getClasses() from Collection to to Stream #1027

### PR DESCRIPTION
**Pull request by JebeKerman**

### CODE REVIEW

File: sootup.tests/src/test/java/sootup/tests/validator/NewValidatorTest.java [hunk 1]
```diff
@@ -61,11 +61,7 @@ public PackageName getPackageName() {
 
     classes = new HashSet<>(); // Set to track the classes to check
 
-    for (SootClass aClass : view.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    view.getClasses().filter(aClass -> !aClass.isLibraryClass()).forEach(classes::add);
```
Comment (swissiety) [inline sootup.tests/src/test/java/sootup/tests/validator/NewValidatorTest.java]: offPR: actually we dont need the filter here

File: sootup.tests/src/test/java/sootup/tests/validator/LocalsValidatorTest.java [hunk 2]
```diff
@@ -59,11 +59,7 @@ public PackageName getPackageName() {
 
     classes = new HashSet<>(); // Set to track the classes to check
 
-    for (SootClass aClass : jimpleView.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    jimpleView.getClasses().filter(aClass -> !aClass.isLibraryClass()).forEach(classes::add);
```
Comment (swissiety) [inline sootup.tests/src/test/java/sootup/tests/validator/LocalsValidatorTest.java]: offPR: same as below

File: sootup.tests/src/test/java/sootup/tests/validator/FieldRefValidatorTest.java [hunk 3]
```diff
@@ -61,11 +61,7 @@ public PackageName getPackageName() {
 
     classes = new HashSet<>(); // Set to track the classes to check
 
-    for (SootClass aClass : jimpleView.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    jimpleView.getClasses().filter(aClass -> !aClass.isLibraryClass()).forEach(classes::add);
```
Comment (swissiety) [inline sootup.tests/src/test/java/sootup/tests/validator/FieldRefValidatorTest.java]: offPR: same as below

File: sootup.tests/src/test/java/sootup/tests/validator/CheckInitValidatorTest.java [hunk 4]
```diff
@@ -60,11 +60,7 @@ public PackageName getPackageName() {
 
     classes = new HashSet<>(); // Set to track the classes to check
 
-    for (SootClass aClass : jimpleView.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    jimpleView.getClasses().filter(aClass -> !aClass.isLibraryClass()).forEach(classes::add);
```
Comment (swissiety) [inline sootup.tests/src/test/java/sootup/tests/validator/CheckInitValidatorTest.java]: offPR: same as below

File: sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java [hunk 5]
```diff
@@ -447,11 +447,7 @@ public CallGraph addClass(@Nonnull CallGraph oldCallGraph, @Nonnull JavaClassTyp
...
     Set<SootClass> classes = new HashSet<>(); /* Set to track the classes to check */
-    for (SootClass aClass : view.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    classes.addAll(view.getClasses().filter(aClass -> !aClass.isLibraryClass()).collect(Collectors.toSet()));
 
     Collection<SootMethod> mainMethods = new HashSet<>(); /* Set to store the methods */
     for (SootClass aClass : classes) {
```
Comment (swissiety) [inline sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java]: next iteration: lets not collect and store them classes in a HashSet - use the stream directly in the loop i.e. we can transform the loop to make use of streams as well :)

**Review by swissiety — COMMENTED** [review]
thx for your work @JebeKerman !

File: sootup.tests/src/test/java/sootup/tests/validator/LocalsValidatorTest.java (hunk 2 above)
Comment (JebeKerman) [reply sootup.tests/src/test/java/sootup/tests/validator/LocalsValidatorTest.java]: I removed the filter

File: sootup.tests/src/test/java/sootup/tests/validator/CheckInitValidatorTest.java (hunk 4 above)
Comment (JebeKerman) [reply sootup.tests/src/test/java/sootup/tests/validator/CheckInitValidatorTest.java]: I removed the filter

File: sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java (hunk 5 above)
Comment (JebeKerman) [reply sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java]: I tried to get rid of the two for loops. I hope this was correct. (First time contributing to open source, so no idea if what I'm doing is correct)

Comment (swissiety) [reply sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java]: That looks good! We just don't need to create the Pair as we can get the type of the declaring class via method -> method.getMethodSignature.getDeclClassType(). Don't be afraid of doing something wrong - we have version control to tackle that ;) So everything is a opportunity to learn and as long as the communication is good a lot of things are possible :)

Comment (swissiety) [reply sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java]: which is not even necessary as we can work on the MethodSubSignature - JavaIdentifierFactory.getInstance().isMainSubSignature( method.getSignature().getSubSignature() ) will do the job of everything that is inside the equals part of the condition

**Review by swissiety — APPROVED** [review]
well done! That simplified a lot.

**Comment by JebeKerman** [comment]
Very cool, thanks for the help :)

**Pull request merged**